ci/build-msys2.sh

```diff
@@ -1,7 +1,9 @@
 #!/bin/sh -e
 
 if [ "$1" = "meson" ]; then
-    meson setup build            \
+    python3.11 -m ensurepip
+    python3.11 -m pip install meson
+    python3.11 -m mesonbuild.mesonmain setup build \
```

Comment:
We don't officially support this `-m` invocation style, but I'm mostly surprised that pip install meson wouldn't place a brand new "meson" command using python 3.11 on $PATH. Usually it's hard to get pip to *stop* installing new scripts, since they would overwrite any that were installed by a different python version. :D

Reply:
I can use meson directly. All this hopefully is only temporary until MSYS2 switches to 3.11, but not sure when it happens and those random crashes are really triggering me.

Reply:
@eli-schwartz Also, by the way, you should support module invocation. I dislike calling some random wrapper scripts / binaries.

Reply:
We've previously arbitrarily renamed the main package from meson to mesonbuild, which was fine because we didn't document it one way or the other. Documenting the mesonbuild.mesonmain API feels a bit like documenting some kind of API...

We definitely don't want to encourage people to try importing our main() function from their application. Which is admittedly not quite the same as `python -m`. Even so, in my experience tools that are invoked with `python -m` are predominantly the ones where the python version matters (for reasons other than "one version of python itself has a segfault bug"), e.g. pip which operates on a python environment itself.

As far as "random wrapper scripts" goes, I'm personally quite a fan of meson's support for building itself as a zipapp `meson.pyz` which you can then stick in `$PATH` or run as `python3.11 meson.pyz --help` and is guaranteed to work.

All this being said I don't personally anticipate module invocation ceasing to work any time soon, documented or not. I'm just unsure whether to feel comfortable committing to that...